From 3e745666c74f6aa2b5c1eae178ee9395628d61bf Mon Sep 17 00:00:00 2001
From: David Eads <deads@redhat.com>
Date: Thu, 29 Oct 2020 13:55:55 +0100
Subject: [PATCH] UPSTREAM: <carry>: hardcoded restmapper with a few entries to
 rebootstrap SDN when SDN is down

UPSTREAM: <carry>: use hardcoded rest mapper from library-go

OpenShift-Rebase-Source: a00f75daeb5
---
 .../k8s.io/apiserver/pkg/server/options/admission.go  |  2 +-
 .../apiserver/pkg/server/options/patch_restmapper.go  | 11 +++++++++++
 2 files changed, 12 insertions(+), 1 deletion(-)
 create mode 100644 staging/src/k8s.io/apiserver/pkg/server/options/patch_restmapper.go

diff --git a/staging/src/k8s.io/apiserver/pkg/server/options/admission.go b/staging/src/k8s.io/apiserver/pkg/server/options/admission.go
index 61085e94ebd..1e9dd4e61c9 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/options/admission.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/options/admission.go
@@ -153,7 +153,7 @@ func (a *AdmissionOptions) ApplyTo(
 	discoveryClient := cacheddiscovery.NewMemCacheClient(kubeClient.Discovery())
 	discoveryRESTMapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
 	genericInitializer := initializer.New(kubeClient, dynamicClient, informers, c.Authorization.Authorizer, features,
-		c.DrainedNotify(), discoveryRESTMapper)
+		c.DrainedNotify(), NewAdmissionRESTMapper(discoveryRESTMapper))
 	initializersChain := admission.PluginInitializers{genericInitializer}
 	initializersChain = append(initializersChain, pluginInitializers...)
 
diff --git a/staging/src/k8s.io/apiserver/pkg/server/options/patch_restmapper.go b/staging/src/k8s.io/apiserver/pkg/server/options/patch_restmapper.go
new file mode 100644
index 00000000000..bce6453ee5f
--- /dev/null
+++ b/staging/src/k8s.io/apiserver/pkg/server/options/patch_restmapper.go
@@ -0,0 +1,11 @@
+package options
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/openshift/library-go/pkg/client/openshiftrestmapper"
+)
+
+func NewAdmissionRESTMapper(delegate meta.RESTMapper) meta.RESTMapper {
+	return openshiftrestmapper.NewOpenShiftHardcodedRESTMapper(delegate)
+}
-- 
2.46.0

